feat[ui]: added extensive list of timezones and cosmetic update to th…

yusrimathews opened the pull request:
…e config modal of the schedule block

Fixes #295 

## Description

1. Enhanced Timezone Support
- **Expanded timezone options**: Added 90+ new timezone options to the schedule block configuration
- **Organized by regions**: Timezones are now grouped logically by geographical regions.
2. UI Improvements
- **Better text wrapping**: Added text-wrap class to the schedule config component to prevent text overflow
- **Improved readability**: Enhanced the display of next run and last run times in the schedule configuration
3. Files Modified
- `schedule-modal.tsx`: Added comprehensive timezone dropdown with 90+ options
- `schedule-config.tsx`: Fixed text wrapping for better UI display
- `schedule.ts`: Updated the block configuration with the expanded timezone list
4. Key Benefits
- **Global accessibility**: Users from any timezone can now easily find their local time
- **Better UX**: Organized timezone list makes it easier to find specific locations
- **Comprehensive coverage**: Includes major business centers and population centers worldwide
- **Consistent formatting**: All timezones follow the same naming convention (City/Region (UTC±X))

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally and in CI (`bun run test`)
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have updated version numbers as needed (if needed)
- [x] I confirm that I have read and agree to the terms outlined in the [Contributor License Agreement (CLA)](./CONTRIBUTING.md#contributor-license-agreement-cla)

## Security Considerations:

- [x] My changes do not introduce any new security vulnerabilities
- [x] I have considered the security implications of my changes

## Additional Information:

<img width="711" height="638" alt="Screenshot 2025-07-28 at 15 59 13" src="https://github.com/user-attachments/assets/4b7f7b68-d661-4fb3-bd20-166822fca3b9" />
<img width="711" height="638" alt="Screenshot 2025-07-28 at 15 59 06" src="https://github.com/user-attachments/assets/5322cabb-8ae8-4289-ad50-f6bca36518d1" />
<img width="441" height="247" alt="Screenshot 2025-07-28 at 15 57 18" src="https://github.com/user-attachments/assets/965e3d2a-87d7-4666-b504-b618b723a47e" />

waleedlatif1 commented:
hey @yusrimathews ! thanks for the contribution. with this many options in the dropdown for timezones, we should consider adding a search to the dropdown as well. preferably, we could re-use another one of the components or add an optional search prop to the dropdown component already used here.

yusrimathews commented:
> hey @yusrimathews ! thanks for the contribution. with this many options in the dropdown for timezones, we should consider adding a search to the dropdown as well. preferably, we could re-use another one of the components or add an optional search prop to the dropdown component already used here.

Updated as suggested - the scrolling issue occurs when using the "command" and "command list" tags, so I am using divs instead. I also added keywords to the list of countries so it can be searched easier.
